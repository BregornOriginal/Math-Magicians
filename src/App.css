/* * {
	margin: 0;
	box-sizing: border-box;
}

h1 {

}
.header {
	display: flex;
}

.header nav {
	display: flex;
} */
