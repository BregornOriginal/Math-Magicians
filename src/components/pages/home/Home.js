// import styles from './Home.module.css';
import LoremIpsum from 'react-lorem-ipsum';

export default function Home() {
  return (
    <main>
      <h2>Welcome to our page!</h2>
      <LoremIpsum p={2} />
    </main>
  );
}
