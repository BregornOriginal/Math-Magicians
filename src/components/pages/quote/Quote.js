import styles from './Quote.module.css';

export default function Quote() {
  return (
    <main>
      <div>
        <p className={styles.quote}>
          Mathematics is not about numbers, equations, computations
          or algorithms: it is about understanding. -William Paul Thurston
        </p>
      </div>
    </main>
  );
}
