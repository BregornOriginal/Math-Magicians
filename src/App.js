import React from 'react';
// import './App.css';
import { Routes, Route } from 'react-router-dom';
import Calculator from './components/pages/calculator/Calculator';
import Home from './components/pages/home/Home';
import Quote from './components/pages/quote/Quote';
import Header from './components/header/Header';

function App() {
  return (
    <div className="main-container">
      <React.StrictMode>
        <Header />
        <Routes>
          <Route path="/" element={<Home />} />
          <Route path="quote" element={<Quote />} />
          <Route path="Calculator" element={<Calculator />} />
        </Routes>
      </React.StrictMode>
    </div>
  );
}

export default App;
