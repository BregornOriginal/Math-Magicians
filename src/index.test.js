import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import App from './App';
import Calculator from './components/pages/calculator/Calculator';
import Home from './components/pages/home/Home';
import Quote from './components/pages/quote/Quote';

describe('Calculator actions', () => {
  render(<Calculator />);
  const key6 = screen.getByText('6');
  const key2 = screen.getByText('2');
  const keyResult = screen.getByText('=');
  const keyPlus = screen.getByText('+');
  const keyX = screen.getByText('x');
  const display = document.querySelector('.calculator-display');

  it('operate (6 + 6) * 22, should be equal to 264', () => {
    fireEvent.click(key6);
    fireEvent.click(keyPlus);
    fireEvent.click(key6);
    fireEvent.click(keyResult);
    expect(display.innerHTML).toBe('12');
    fireEvent.click(keyX);
    fireEvent.click(key2);
    fireEvent.click(key2);
    fireEvent.click(keyResult);
    expect(display.innerHTML).toBe('264');
  });
});

describe('Snapshot test', () => {
  const Container = () => {
    <BrowserRouter>
      <App />
    </BrowserRouter>;
  };
  it('test App should render and match the Snapshot', () => {
    const tree = renderer.create(<Container />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render Calculator', () => {
    const AppContainer = () => {
      <BrowserRouter>
        <Calculator />
      </BrowserRouter>;
    };
    const testCalculator = renderer.create(<AppContainer />).toJSON();
    expect(testCalculator).toMatchSnapshot();
  });

  it('should render Home', () => {
    const HomeContainer = () => {
      <BrowserRouter>
        <Home />
      </BrowserRouter>;
    };
    const homeCalculator = renderer.create(<HomeContainer />).toJSON();
    expect(homeCalculator).toMatchSnapshot();
  });

  it('should render Quote', () => {
    const QuoteContainer = () => {
      <BrowserRouter>
        <Quote />
      </BrowserRouter>;
    };
    const quoteCalculator = renderer.create(<QuoteContainer />).toJSON();
    expect(quoteCalculator).toMatchSnapshot();
  });

});
