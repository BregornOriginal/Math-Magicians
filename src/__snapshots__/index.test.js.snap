// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test should render Calculator 1`] = `null`;

exports[`Snapshot test should render Home 1`] = `null`;

exports[`Snapshot test should render Quote 1`] = `null`;

exports[`Snapshot test test App should render and match the Snapshot 1`] = `null`;
